src/manager.wsgi

```diff
@@ -485,8 +486,21 @@ def application(environ, start_response):
     elif filename and filename == "__custom__":
         POST = request.POST
 
-        if "custom_url" in POST and len(POST["custom_url"]) == 0:
-            return response(start_response, "400 Bad Request", _("URL cannot be NULL string"))
+        if "custom_url" in POST:
+            url = POST["custom_url"]
+            if not url:
+                return response(start_response, "400 Bad Request", _("URL cannot be NULL string"))
+            if url.startswith("/") or url.startswith("file:///"):
+                if url.startswith("file://"):
+                    path = Path(urllib.parse.unquote(url[7:]))
+                else:
+                    path = Path(url)
+
+                # Check the file exists and we can open it for binary read
+                try:
+                    open(path, "rb")
```

Comment:
Looks fine overall. Only one nitpick: The way this line is written now is basically a resource leak. I suggest wrapping it in a dummy `with`, such as

```suggestion
                    with open(path, "rb"):
                        # We only check if the file can be opened, we don't read it yet.
                        pass
```

Reply:
Hmmm, I thought the try/except would insulate the leak but maybe I'm wrong there?  I can fix this up.